test/integration/scan_test.go

```diff
@@ -509,3 +511,39 @@ func TestFailedScanWithWrongPreset(t *testing.T) {
 	err, _ := executeCommand(t, args...)
 	assertError(t, err, "scan did not complete successfully")
 }
+
+func retrieveResultsFromScanId(t *testing.T, scanId string) (wrappers.ScanResultsCollection, error) {
+	resultsArgs := []string{
+		"results",
+		"show",
+		flag(params.ScanIDFlag),
+		scanId,
+	}
+	executeCmdNilAssertion(t, "Getting results should pass", resultsArgs...)
+	file, err := ioutil.ReadFile("cx_result.json")
+	results := wrappers.ScanResultsCollection{}
+	_ = json.Unmarshal([]byte(file), &results)
+	os.Remove("cx_result.json")
```

Comment:
If the function fails before removing the file it will be left dangling. Consider doing
`defer func() { os.Remove("cx_result.json")` immediately after executing the results command. Deferring the call will make sure it runs at the end of the function regardless of exit point.